modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/action/ViewFileEntryContentDashboardItemAction.java

```diff
@@ -96,26 +95,33 @@ private String _getViewURL(Locale locale, ThemeDisplay themeDisplay) {
 
 			clonedThemeDisplay.setScopeGroupId(_fileEntry.getGroupId());
 
-			return Optional.ofNullable(
+			String friendlyURL =
 				_assetDisplayPageFriendlyURLProvider.getFriendlyURL(
 					FileEntry.class.getName(), _fileEntry.getFileEntryId(),
-					locale, clonedThemeDisplay)
-			).map(
-				url -> {
-					String backURL = ParamUtil.getString(
-						_httpServletRequest, "backURL");
-
-					if (Validator.isNotNull(backURL)) {
-						return HttpComponentsUtil.setParameter(
-							url, "p_l_back_url", backURL);
-					}
-
-					return HttpComponentsUtil.setParameter(
-						url, "p_l_back_url", themeDisplay.getURLCurrent());
-				}
-			).orElse(
-				StringPool.BLANK
-			);
+					locale, clonedThemeDisplay);
+
+			if (friendlyURL == null) {
+				return StringPool.BLANK;
+			}
+
+			String url = null;
+
+			String backURL = ParamUtil.getString(
+				_httpServletRequest, "backURL");
+
+			if (Validator.isNotNull(backURL)) {
+				url = HttpComponentsUtil.setParameter(
+					friendlyURL, "p_l_back_url", backURL);
+			}
+
+			url = HttpComponentsUtil.setParameter(
```

Comment:
@slnn This is wrong. The original logic just returns when `Validator.isNotNull(backURL)`, but your logic will make the value in the if-block never used.

Reply:
oh, got it, my bad....